### PR TITLE
Better batching for higher fixed costs.

### PR DESCRIPTION
**Pull request by robertwb**
Make Python's BatchElements more aggressive batching in some cases, capping at 10 second rather than 1 second batches by default and excluding fixed cost in this computation.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @iindyk

**Comment by robertwb** [comment]
Failures due to `fatal: write error: No space left on device`

File: sdks/python/apache_beam/transforms/util.py [hunk 1]
```diff
@@ -502,8 +502,11 @@ def _calculate_next_batch_size(self):
     target = self._max_batch_size
 
     if self._target_batch_duration_secs:
-      # Solution to a + b*x = self._target_batch_duration_secs.
-      target = min(target, (self._target_batch_duration_secs - a) / b)
+      # Solution to b*x = self._target_batch_duration_secs.
+      # We ignore the fixed cost in this computation as it has negligeabel
```
Comment (iindyk) [inline sdks/python/apache_beam/transforms/util.py:506]: I think this is wrong: what if the element overhead and fixed cost are both large?

Comment (robertwb) [reply sdks/python/apache_beam/transforms/util.py:506]: In that case we still won't batch (much or at all). That's the problem with absolute limits (but it felt too risky of a backwards incompatible change to remove it altogether). But it increases the range that the more general overhead calculation takes precedence instead.

**Review by iindyk — APPROVED** [review]

**Pull request merged**